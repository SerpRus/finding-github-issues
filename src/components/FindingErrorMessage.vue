<template>
  <div class="finding-error-message" v-if="isErrorMessage">
    {{(message) ? message : 'Nothing found'}}
  </div>
</template>

<script>
export default {
  name: 'FindingErrorMessage',
  props: {
    isErrorMessage: Boolean,
    message: String,
  },
};
</script>

<style lang="scss" scoped>
.finding-error-message {
  text-align: center;
}
</style>
