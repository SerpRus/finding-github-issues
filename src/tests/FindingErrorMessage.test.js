import {mount} from '@vue/test-utils';
import FindingErrorMessage from '../components/FindingErrorMessage.vue';

describe('FindingErrorMessage tests', () => {
  it('FindingErrorMessage isErrorMessage: true with message', () => {
    const wrapper = mount(FindingErrorMessage, {
      props: {
        isErrorMessage: true,
        message: 'test1',
      },
    });

    expect(wrapper.text()).toBe('test1');
  });

  it('FindingErrorMessage isErrorMessage: true no message', () => {
    const wrapper = mount(FindingErrorMessage, {
      props: {
        isErrorMessage: true,
      },
    });

    expect(wrapper.text()).toBe('Nothing found');
  });

  it('FindingErrorMessage isErrorMessage: false', () => {
    const wrapper = mount(FindingErrorMessage, {
      props: {
        isErrorMessage: false,
        message: 'test1',
      },
    });

    expect(wrapper.html()).toBe('<!--v-if-->');
  });

  it('FindingErrorMessage isErrorMessage: true snapshot', () => {
    const wrapper = mount(FindingErrorMessage, {
      props: {
        isErrorMessage: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('FindingErrorMessage isErrorMessage: false snapshot', () => {
    const wrapper = mount(FindingErrorMessage, {
      props: {
        isErrorMessage: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
})
