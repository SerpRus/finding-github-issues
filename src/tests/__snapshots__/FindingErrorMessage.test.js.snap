// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindingErrorMessage tests FindingErrorMessage isErrorMessage: false snapshot 1`] = `"<!--v-if-->"`;

exports[`FindingErrorMessage tests FindingErrorMessage isErrorMessage: true snapshot 1`] = `"<div class=\\"finding-error-message\\">Nothing found</div>"`;
